### PR TITLE
OG tags: avoid PHP warning when og:description is not set

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

In some scenarios (no post excerpt, post with a Premium block), you may have no og:description, thus causing a warning.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

- Start on a WoA site using Jetpack, where you've inserted a Premium Content block into one of your posts.
- Activate the Social plugin.
- Visit that post, and see the PHP warning in your logs.
- Apply this branch; it should help.

### CODE REVIEW

**Comment by danielpost** [comment]
Thanks! I'm going to play around with this a bit. I think there's some flawed logic in [the block right above it](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/social/src/class-meta-tags.php#L267). It appears that we should always call `get_description` since it provides a fallback if there's no description set.

**Comment by jeherve** [comment]
> Thanks! I'm going to play around with this a bit. I think there's some flawed logic in [the block right above it](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/social/src/class-meta-tags.php?rgh-link-date=2022-08-23T11%3A10%3A40Z#L267). It appears that we should always call `get_description` since it provides a fallback if there's no description set.

Thanks! Feel free to take the PR from me if that helps :)

**Review by anomiex — DISMISSED** [review]
Looks good, straightforward addition of a check. Haven't tested it myself though.

**Comment by danielpost** [comment]
@jeherve I've tweaked it a bit and added some more tests, let me know what you think 🙂

**Review by jeherve — COMMENTED** [review]
This looks good to me, but I opened the PR so cannot approve it. I'll let someone else take a look at it :)

**Review by anomiex — APPROVED** [review]
Changes seem ok to me. Haven't tested.

**Pull request merged**
